qlib/workflow/online/strategy.py

```diff
@@ -7,135 +7,87 @@
-            need_log (bool, optional): print log or not. Defaults to True.
         """
         self.name_id = name_id
-        self.trainer = trainer
         self.logger = get_module_logger(self.__class__.__name__)
-        self.need_log = need_log
         self.tool = OnlineTool()
 
-    def prepare_signals(self, delay: bool = False):
-        """
-        After perparing the data of last routine (a box in box-plot) which means the end of the routine, we can prepare trading signals for next routine.
-
-        NOTE: Given a set prediction, all signals before these prediction end time will be prepared well.
-
-        Args:
-            delay: bool
-                If this method was called by `delay_prepare`
-        """
-        raise NotImplementedError(f"Please implement the `prepare_signals` method.")
-
-    def prepare_tasks(self, *args, **kwargs):
+    def prepare_tasks(self, cur_time, **kwargs) -> List[dict]:
         """
-        After the end of a routine, check whether we need to prepare and train some new tasks.
+        After the end of a routine, check whether we need to prepare and train some new tasks based on cur_time (None for latest)..
         Return the new tasks waiting for training.
 
         You can find last online models by OnlineTool.online_models.
         """
         raise NotImplementedError(f"Please implement the `prepare_tasks` method.")
 
-    def prepare_online_models(self, tasks, check_func=None, **kwargs):
+    def prepare_online_models(self, models, cur_time=None, check_func=None) -> List[object]:
         """
-        Use trainer to train a list of tasks and set the trained model to `online`.
+        A typically implementation, but maybe you will need old models by online_tool.
+        Select some models as the online models from the trained models.
 
-        NOTE: This method will first offline all models and online the online models prepared by this method. So you can find last online models by OnlineTool.online_models if you still need them.
+        NOTE: This method offline all models and online the online models prepared by this method (if have). So you can find last online models by OnlineTool.online_models if you still need them.
 
         Args:
             tasks (list): a list of tasks.
             check_func: the method to judge if a model can be online.
                 The parameter is the model record and return True for online.
                 None for online every models.
-            **kwargs: will be passed to end_train which means will be passed to customized train method.
 
+        Returns:
+            List[object]: a list of selected models.
         """
-        if check_func is None:
-            check_func = lambda x: True
-        online_models = []
-        if len(tasks) > 0:
-            new_models = self.trainer.train(tasks, **kwargs)
-            for model in new_models:
-                if check_func(model):
+        if check_func is not None:
```

Comment:
It's hard to come up with a scenario for the scenario `check_func`. 
I think we could simplify this method.